presto-main/src/main/java/com/facebook/presto/server/security/OpenLdapBinder.java

```diff
@@ -13,32 +13,60 @@
  */
 package com.facebook.presto.server.security;
 
+import com.google.common.base.Throwables;
+import io.airlift.log.Logger;
+
 import javax.inject.Inject;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
 import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+
+import java.util.Optional;
 
+import static com.google.common.base.Preconditions.checkState;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 public class OpenLdapBinder
         implements LdapBinder
 {
-    private final String baseDistinguishedName;
+    private static final Logger LOG = Logger.get(ActiveDirectoryBinder.class);
+    private final String userObjectClass;
+    private final Optional<String> baseDistinguishedName;
 
     @Inject
     public OpenLdapBinder(LdapServerConfig config)
     {
-        baseDistinguishedName = requireNonNull(config.getBaseDistinguishedName(), "baseDistinguishedName is null");
+        userObjectClass = requireNonNull(config.getUserObjectClass(), "userObjectClass is null");
+        baseDistinguishedName = Optional.ofNullable(config.getBaseDistinguishedName());
     }
 
     @Override
     public String getBindDistinguishedName(String user)
     {
-        return format("uid=%s,%s", user, baseDistinguishedName);
+        return format("uid=%s,%s", user, baseDistinguishedName.get());
     }
 
     @Override
     public boolean checkForGroupMembership(String user, String groupDistinguishedName, DirContext context)
     {
-        throw new UnsupportedOperationException("Group membership based authorization not yet supported");
+        checkState(baseDistinguishedName.isPresent(), "Base distinguished name (DN) for user %s is missing", user);
```

Comment:
looks like this is the same code as the check in AD? Do you think we can just move it to the LdapFilter?